### PR TITLE
MAINT Renormalizing FFT correctly

### LINKED ISSUE #386: Inverse fft in 2D backend
EDIT by EO:
- [x] Using the appropriate normalization for the ifft (ie both backend files)
- [x] Adding a unit test in 2d/test for testing the fft, which is not currently tested 
- [x] Normalizing the wavelet in the Fourier domain (ie the file filter_bank.py) to avoid the new code to break and satisfy the previous unit tests
NOTE: this might lead to a small(not to say tiny) speed-up, because the dot-wise multiplication usually use quite some resources!

In 2D backend code, in
`            output = torch.ifft(input, 2, normalized=False)*input.size(-2)*input.size(-3)
`
why is the inverse fft routine multiplied by the term *input.size(-2)*input.size(-3)? I was expecting that transforming a matrix back and forth to the Fourier domain give the same result. In other words, I was expecting outputs of the following test code match.

```
import torch
from kymatio.scattering2d import backend as bd2

pad = bd2.Pad(2 ** 0, pre_pad=False)

# -- parameters
M = 8
N = 8
L = 8
J = 4

# -- synthesise data
y = torch.rand(1, 1, M, N)
for i in range(M):
    for j in range(N):
        y[:,:,i,j] = i+j

y_pad = pad(y)                                    # pad to add complex dimension
y_fft = bd2.fft(y_pad, 'C2C')                     # fft
y_fft_inv = bd2.fft(y_fft, 'C2C', inverse=True)   # inverse fft
y_fft_inv_un = y_fft_inv[..., 1:-1, 1:-1, 0]      # unpad

# -- print
print(y)
print(y_fft_inv_un)
```

### PR DESCRIPTION
**Pull request by edouardoyallon**
remove inappropriate renormalization

Close #386

### CODE REVIEW

File: kymatio/scattering2d/tests/test_torch_scattering2d.py [hunk 1]
```diff
@@ -309,6 +309,8 @@ def test_fft(self, backend):
         y[1, 0, :] = x[0, 0, :] + x[0, 1, :] - x[1, 0, :] - x[1, 1, :]
         y[1, 1, :] = x[0, 0, :] - x[0, 1, :] - x[1, 0, :] + x[1, 1, :]
 
+        y = y/2.0 # renormalization
```
Comment (MuawizChaudhary) [inline kymatio/scattering2d/tests/test_torch_scattering2d.py]: Why divide by 2?

File: kymatio/scattering2d/filter_bank.py [hunk 2]
```diff
@@ -63,7 +61,7 @@ def filter_bank(M, N, J, L=8):
         phi_signal_fourier_res = periodize_filter_fft(phi_signal_fourier, res)
         filters['phi'][res] = phi_signal_fourier_res
         # Normalization to avoid doing it with the FFT.
-        filters['phi'][res] /= M*N // 2 ** (2 * J)
+        filters['phi'][res] *= M*N // 2 ** (2 * J)
```
[the remark this answers is not in the document]
Comment (edouardoyallon) [reply kymatio/scattering2d/filter_bank.py]: this is the "true" renormalization

File: kymatio/scattering2d/tests/test_torch_scattering2d.py (hunk 1 above)
Comment (edouardoyallon) [reply kymatio/scattering2d/tests/test_torch_scattering2d.py]: 2^2=4, the FFT is unitary, and now the matrix y=Ax is unitary!

Comment (MuawizChaudhary) [reply kymatio/scattering2d/tests/test_torch_scattering2d.py]: Never mind, reading the pytorch documentation the result of the FFT is normalized by dividing by the square root of the product of the first two dimensions.

Comment (janden) [reply kymatio/scattering2d/tests/test_torch_scattering2d.py]: Can we make that clear? E.g. `y /= torch.sqrt(2 ** 2)`. It's a little silly, but more clear, I think.

File: kymatio/scattering2d/filter_bank.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (janden) [reply kymatio/scattering2d/filter_bank.py]: You still haven't explained why the line above is removed while this one is changed. Also, spaces around asterisk.

File: kymatio/scattering2d/backend/torch_backend.py [hunk 3]
```diff
@@ -223,12 +223,12 @@ def fft(x, direction='C2C', inverse=False):
         raise RuntimeError('Tensors must be contiguous.')
 
     if direction == 'C2R':
-        output = torch.irfft(x, 2, normalized=False, onesided=False) * x.shape[-2] * x.shape[-3]
+        output = torch.irfft(x, 2, normalized=True, onesided=False)
```
Comment (janden) [inline kymatio/scattering2d/backend/torch_backend.py]: The unitary FFT, really? This is quite non-standard. Why not stick to the regular ones?

Comment (edouardoyallon) [reply kymatio/scattering2d/backend/torch_backend.py]: well, the dotwise multiplication is anyway slow - i'm also ok to remove the "normalized" fft

File: kymatio/scattering2d/filter_bank.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (edouardoyallon) [reply kymatio/scattering2d/filter_bank.py]: a normalization is due because we downsample

**Comment by edouardoyallon** [comment]
It seems this catched a bigger bug: the numpy frontend/backend ran into an error, I'll correct and update!

File: kymatio/scattering2d/filter_bank.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (janden) [reply kymatio/scattering2d/filter_bank.py]: And the ψ filters are not downsampled?

File: kymatio/scattering2d/backend/torch_backend.py (hunk 3 above)
Comment (janden) [reply kymatio/scattering2d/backend/torch_backend.py]: Should be faster then with `normalized=False` since that's only one elementwise multiplication (in the IFFT).

File: kymatio/scattering2d/tests/test_torch_scattering2d.py (hunk 1 above)
Comment (edouardoyallon) [reply kymatio/scattering2d/tests/test_torch_scattering2d.py]: This is now fixed(plz resolve if OK)
